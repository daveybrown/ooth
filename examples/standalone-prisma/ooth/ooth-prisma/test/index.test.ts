require('dotenv').config({ path: './.env.test' });
import { cloneDeep } from 'lodash';
import { OothPrisma } from '../src';
const { Prisma } = require('../generated/prisma-client');

const obfuscate = (obj: any, ...paths: string[]) => {
  const res = cloneDeep(obj);
  for (const path of paths) {
    const keys = path.split('.');
    let current = res;
    for (const key of keys.slice(0, -1)) {
      current = current[key];
    }
    current[keys[keys.length - 1]] = '<obfuscated>';
  }

  return res;
};

let prisma: any = {};
let oothPrisma: any = {};


describe('ooth-prisma', () => {
  beforeAll(async () => {
    prisma = new Prisma();
    oothPrisma = new OothPrisma(prisma); 
  });

  afterAll(async () => {

  });

  afterEach(async () => {

  });

  test('can insert user and get it', async () => {
    const id = await oothPrisma.insertUser({
      foo: {
        baz: 'bar',
      },
    });
    expect(typeof id).toBe('string');
    const user = await oothPrisma.getUserById(id);
    expect(obfuscate(user, '_id')).toMatchSnapshot();
  });

  test.skip('can update user', async () => {
    const id = await oothPrisma.insertUser({
      foo: {
        baz: 'bar',
      },
    });
    await oothPrisma.updateUser(id, {
      foo: {
        baz: 'bar',
        baz2: 'bar2',
      },
      foo2: {
        baz: 'bar2',
      },
    });
    const user = await oothPrisma.getUserById(id);
    expect(obfuscate(user, '_id')).toMatchSnapshot();
  });

  test('can get user by value', async () => {
    await oothPrisma.insertUser({
      foo: {
        bar: 'baz',
      },
      foo2: {
        bar2: 'baz2',
      },
    });
    const user = await oothPrisma.getUserByValue(['foo.bar', 'foo2.bar2'], 'baz2');
    expect(obfuscate(user, '_id')).toMatchSnapshot();
  });

});
