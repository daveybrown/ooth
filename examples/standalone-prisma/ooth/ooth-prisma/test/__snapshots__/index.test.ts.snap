// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ooth-prisma can update user 1`] = `
Object {
  "_id": "<obfuscated>",
  "foo": Object {
    "baz": "bar",
    "baz2": "bar2",
  },
  "foo2": Object {
    "baz": "bar2",
  },
  "foo3": Object {
    "baz": "bar3",
  },
}
`;
